test/0.8.9/sepolia-deposit-adapter.test.js

```diff
@@ -0,0 +1,81 @@
+const { contract, ethers } = require('hardhat')
+const { assert } = require('../helpers/assert')
+
+const { EvmSnapshot } = require('../helpers/blockchain')
+
+contract('SepoliaDepositAdapter impl', ([deployer]) => {
+  let depositAdapter
+  let snapshot
+  let bepoliaToken
+  // const sepoliaDepositAdapterContract = '0x899e45316FaA439200b36c7d7733192530e3DfC0'
+  const sepoliaDepositContract = '0x7f02C3E3c98b133055B8B348B2Ac625669Ed295D'
+  const bepoliaTokenHolder = '0x388Ea662EF2c223eC0B047D41Bf3c0f362142ad5'
+  // const EOAddress = '0x6885E36BFcb68CB383DfE90023a462C03BCB2AE5'
+
+  before('deploy lido with dao', async () => {
+    // depositAdapter = await SepoliaDepositAdapter.at(sepoliaDepositAdapterContract)
+    // depositAdapter = await SepoliaDepositAdapter.new(deployer, [sepoliaDepositContract], { from: deployer })
+    depositAdapter = await ethers.deployContract('SepoliaDepositAdapter', [sepoliaDepositContract])
+    console.log('depositAdapter address', depositAdapter.address)
+
+    const depositAdapterVersion = await depositAdapter.VERSION()
+    assert.equals(depositAdapterVersion, 2)
+
+    bepoliaToken = await ethers.getContractAt('SepoliaDepositContract', sepoliaDepositContract)
+
+    snapshot = new EvmSnapshot(ethers.provider)
+    await snapshot.make()
+  })
+
+  afterEach(async () => {
+    await snapshot.rollback()
+  })
+
+  describe('SepoliaDepositAdapter Logic', () => {
+    it(`transfer Bepolia tokens`, async () => {
```

Comment:
Looks like it should be in 'before'?

Reply:
Well. that could be but it's not required for every test

Reply:
Ok. Then, it could be a separate context just for the test that requires a token with its own before hook. 

Now, it looks like we're testing an impersonated version of a token, which we should not.

Reply:
We are impersonating the account with tokens. Account is our EOA and not the same as temp account spinning with hardhat for tests. So, it's not possible to transfer tokens from EOA to contract or did I miss something?